### PR TITLE
chore(eslint): upgrade eslint

### PR DESCRIPTION
**Pull request by Asalem1**
Upgraded ESLINT and updated linter errors

https://github.com/typescript-eslint/typescript-eslint/issues/389

### CODE REVIEW

**Comment by 121watts** [comment]
Pretty massive.  Apologies.  Mostly unneeded `awaits`

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -1,62 +1,69 @@
...
-  "parser": "@typescript-eslint/parser",
-  "parserOptions": {
-    "project": "./tsconfig.json",
-    "tsconfigRootDir": __dirname,
-    "ecmaFeatures": {
-      "jsx": true
-    }
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: './tsconfig.test.json',
```
Comment (drdelambre) [inline ui/.eslintrc.js]: 😬 sorry

File: ui/src/Signin.tsx [hunk 2]
```diff
@@ -50,7 +50,9 @@ export class Signin extends PureComponent<Props, State> {
 
     if (this.hasMounted) {
       this.setState({loading: RemoteDataState.Done})
-      this.intervalID = setInterval(this.checkForLogin, FETCH_WAIT)
+      this.intervalID = setInterval(() => {
+        this.checkForLogin()
```
Comment (drdelambre) [inline ui/src/Signin.tsx]: this is better!

File: ui/src/clientLibraries/graphics/GoLogo.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Libraries
 import React, {SFC} from 'react'
 
+/* eslint-disable no-mixed-spaces-and-tabs */
```
Comment (drdelambre) [inline ui/src/clientLibraries/graphics/GoLogo.tsx]: these should have a matching `/* eslint-enable */` at the end of the file, in the name of compositional ease (changing include style, copy pasta, etc)

File: ui/src/clientLibraries/graphics/JSLogo.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Libraries
 import React, {SFC} from 'react'
 
+/* eslint-disable no-mixed-spaces-and-tabs */
```
Comment (drdelambre) [inline ui/src/clientLibraries/graphics/JSLogo.tsx]: needs a matching `/* eslint-enable */`

File: ui/src/clientLibraries/graphics/PythonLogo.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 // Libraries
 import React, {SFC} from 'react'
 
+/* eslint-disable no-mixed-spaces-and-tabs */
```
Comment (drdelambre) [inline ui/src/clientLibraries/graphics/PythonLogo.tsx]: needs a matching `/* eslint-enable */`

File: ui/src/dashboards/components/dashboard_index/DashboardCard.tsx [hunk 6]
```diff
@@ -154,35 +154,27 @@ class DashboardCard extends PureComponent<Props> {
...
-      await this.props.onCreateLabel(label.name, label.properties)
-
-      // notify success
-    } catch (err) {
-      console.error(err)
-      // notify of fail
-      throw err
-    }
+  private handleCreateLabel = async (label: Label) => {
+    await this.props.onCreateLabel(label.name, label.properties) // eslint-disable-line
```
Comment (drdelambre) [inline ui/src/dashboards/components/dashboard_index/DashboardCard.tsx]: what went on here?

File: ui/src/external/codemirror.ts [hunk 7]
```diff
@@ -6,12 +6,12 @@ import {
 } from 'src/shared/constants/codeMirrorModes'
 import 'codemirror/addon/hint/show-hint'
 
-/* tslint:disable */
+/* eslint-disable */
```
Comment (drdelambre) [inline ui/src/external/codemirror.ts]: needs a matching `/* eslint-enable */`

File: ui/src/labels/components/CreateLabelOverlay.tsx [hunk 8]
```diff
@@ -94,10 +94,9 @@ class CreateLabelOverlay extends Component<Props, State> {
 
     try {
       onCreateLabel(this.state.label)
-      // clear form on successful submit
-      this.resetForm()
     } finally {
       onDismiss()
+      this.resetForm()
```
Comment (drdelambre) [inline ui/src/labels/components/CreateLabelOverlay.tsx]: is this a bug on error?

File: ui/src/shared/components/KeyboardShortcuts.test.tsx [hunk 9]
```diff
@@ -13,6 +13,7 @@ const setup = (override?) => {
     ...override,
   }
 
+  // eslint-disable-next-line
```
Comment (drdelambre) [inline ui/src/shared/components/KeyboardShortcuts.test.tsx]: is there a specific rule we can add here?

File: ui/src/templates/utils/index.ts [hunk 10]
```diff
@@ -26,10 +26,10 @@ export function findIncludedsFromRelationships<
 export function findIncludedFromRelationship<
   T extends {id: string; type: TemplateType}
 >(
-  includeds: {id: string; type: TemplateType}[],
+  included: {id: string; type: TemplateType}[],
```
Comment (drdelambre) [inline ui/src/templates/utils/index.ts]: 😲

File: ui/src/types/codemirror.ts [hunk 11]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (drdelambre) [inline ui/src/types/codemirror.ts]: needs a matching `/* eslint-enable */`

File: ui/src/types/promises.ts [hunk 12]
```diff
@@ -4,8 +4,8 @@ export interface CancelBox<T> {
 }
 
 export class CancellationError extends Error {
-  constructor() {
-    super(...arguments)
+  constructor(...args) {
+    super(...args)
```
Comment (drdelambre) [inline ui/src/types/promises.ts]: 👍

File: ui/src/variables/components/VariableFormContext.tsx [hunk 13]
```diff
@@ -73,7 +73,7 @@ class VariableFormContext extends PureComponent<Props> {
     return <VariableForm {...props} />
   }
 
-  private handleHideOverlay() {
+  private handleHideOverlay = () => {
```
Comment (drdelambre) [inline ui/src/variables/components/VariableFormContext.tsx]: doesnt this lose props when it goes through the transpiler?

Comment (121watts) [reply ui/src/variables/components/VariableFormContext.tsx]: nope 😸

Comment (drdelambre) [reply ui/src/variables/components/VariableFormContext.tsx]: that's super weird, cause it's binding to a context that is defined.... when.... I gotta go rebuild my mental model.

File: ui/cypress/e2e/buckets.test.ts [hunk 14]
```diff
@@ -54,17 +54,19 @@ describe('Buckets', () => {
...
-      cy.getByTestID(`bucket--card--name ${bucket1}`).should('exist')
-
       cy.getByTestID(`context-delete-menu ${bucket1}`).click()
       cy.getByTestID(`context-delete-bucket ${bucket1}`).click()
 
+      // normally we would assert for empty state here
+      // but we cannot because of the default system buckets
+      // since cypress selectors are so fast, that sometimes a bucket
+      // that is deleted will be selected before it gets deleted
+      cy.reload()
```
Comment (ebb-tide) [inline ui/cypress/e2e/buckets.test.ts]: got it.

File: ui/tsconfig.test.json [hunk 15]
```diff
@@ -4,7 +4,16 @@
     "assets",
     "build",
     "node_modules",
-    "cypress",
-    "webpack.tsconfig.json"
+    "webpack.tsconfig.json",
+    "webpack.*.ts",
+    "coverage"
```
Comment (drdelambre) [inline ui/tsconfig.test.json]: 👍

**Comment by drdelambre** [comment]
GOOD LUCK WITH YOUR MERGE CONFLICTS EVERYONE

**Pull request merged**